packages/common/src/util.js

```diff
@@ -788,3 +789,51 @@ function promisifyAll(Class, options) {
 }
 
 util.promisifyAll = promisifyAll;
+
+/**
+ * Determines a service base URL using the following precendence rule:
+ * 1 - Property 'apiEndpoint' in the Service configuration object.
+ * 2 - Environment variables.
+ * 3 - Default endpoint.
+ *
+ * @param {object=} options - Service configuration object.
+ * @param {array} environmentVariables - Array of accepted environment
+ *     variables through which a custom api endpoint can be specified.
+ * @param {string=} defaultApiEndpoint - Fallback api endpoint to use.
+ * @param {boolean} trimProtocol - If true, trim the leading trimProtocol.
+ *     Useful for grcp based service endpoints.
+ * @return {object=} base - Object containg the baseUrl and a boolean
+ *     property indicating whether the baseUrl is a customEndpoint or not.
+ * */
+function determineBaseUrl(options, environmentVariables,
```

Comment:
I think this can be moved onto `Service.prototype.determineBaseUrl_`.

Reply:
What about services that rely on gapic? These don't extend from services

Reply:
Gotcha, this is fine then :)